subsys/bluetooth/controller/multithreading_lock.h

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright (c) 2019 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+ */
+
+/**
+ * @file multithreading_lock.h
+ *
+ * @brief APIs for ensuring BLE controller threadsafe operation.
+ */
+
+#ifndef MULTITHREADING_LOCK_H__
+#define MULTITHREADING_LOCK_H__
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#include <zephyr.h>
+
+/** Macro for acquiring a lock */
+#define MULTITHREADING_LOCK_ACQUIRE() \
+	multithreading_lock_acquire(K_FOREVER)
+
+/** Macro for acquiring a lock without waiting. */
+#define MULTITHREADING_LOCK_ACQUIRE_NO_WAIT() \
+	multithreading_lock_acquire(K_NO_WAIT)
+
+/** Macro for acquiring a lock while waiting forever. */
+#define MULTITHREADING_LOCK_ACQUIRE_FOREVER_WAIT() \
+	multithreading_lock_acquire(K_FOREVER)
+
+/** Macro for releasing a lock */
+#define MULTITHREADING_LOCK_RELEASE() multithreading_lock_release()
+
+
+/** @brief Try to take the lock with the specified blocking behavior.
+ *
+ * This API call will be blocked for the time specified by @ref
+ * CONFIG_BLECTLR_THREADSAFETY_LOCK_TIMEOUT_MSEC and then return error code.
+ *
+ * @param[in] timeout     Timeout value (in milliseconds) for the locking API.
+ *
+ * @retval 0              Success
+ * @retval - ::NRF_EBUSY  Returned without waiting.
```

Comment:
I doubt `k_sem_take` returns `NRF_BUSY` unless we are very lucky

Reply:
fixed